feat(audit): check user group memberships

HastD opened the pull request:
Membership in groups like `docker` or `libvirt` grants the user access equivalent to root. People sometimes add their main user account to these groups for convenience, possibly without being aware of the security implications. This adds a new check to the audit script that flags membership in these groups and informs the user.

Additionally, the `systemd-journal` group is flagged as a warning due to allowing viewing the kernel logs, and other unrecognized groups are flagged with a generic warning about the potential security risks of group memberships.

The only groups that aren't flagged are the user's username group, `wheel` (which is already checked earlier), and `usbguard`.

TheHooly commented:
The only groups a user account would be expected to be in are their own user group, wheel if it is an administrative account, and usbguard if they use the Gnome integration.

The simplest way would be to treat any unexpected groups as failures, and it also avoids falling into the badness enumeration approach.

HastD committed:
feat(audit): check user group memberships

Membership in groups like `docker` or `libvirt` grants the user access
equivalent to root. People sometimes add their main user account to
these groups for convenience, possibly without being aware of the
security implications. This adds a new check to the audit script that
flags membership in these groups and informs the user.

Additionally, the `systemd-journal` group is flagged as a warning due to
allowing viewing the kernel logs, and other unrecognized groups are
flagged with a generic warning about the potential security risks of
group memberships.

The only groups that aren't flagged are the user's username group,
`wheel` (which is already checked earlier), and `usbguard`.

RoyalOughtness committed:
Merge branch 'live' into audit-groups
